x/poe/keeper/keeper.go

```diff
@@ -40,29 +42,43 @@ func NewKeeper(
 	}
 
 	return Keeper{
-		codec:       marshaler,
-		storeKey:    key,
-		paramStore:  paramSpace,
-		twasmKeeper: twasmK,
+		codec:             marshaler,
+		storeKey:          key,
+		paramStore:        paramSpace,
+		twasmKeeper:       twasmK,
+		contractAddrCache: make(map[types.PoEContractType]sdk.AccAddress),
```

Comment:
nit: you could use `len(PoEContractType_name)-1` to init the map. `UNDEFINED` does not exist and we have exactly this number of contracts